File: 06_Algorithm/leetcode_724.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# File              : leetcode_724.py
# Author            : Andy
# Date              : 31.07.2022
# Last Modified Date: 31.07.2022
# Last Modified By  : Andy

# 主体思路就是当前索引左边的前缀和，与总值减去当前前缀和
# 两者相等时返回当前索引即可

class Solution:
    def pivotIndex(self, nums):
        total = sum(nums)
        sums = 0
        for i in range(len(nums)):
            sums += nums[i]
            if sums - nums[i] == total - sums:
                return i
        return -1

File: Algorithm/leetcode_724.go
/**
 * File              : leetcode_724.go
 * Author            : Andy
 * Date              : 31.07.2022
 * Last Modified Date: 31.07.2022
 * Last Modified By  : Andy
 */

// 主体思路：左侧前缀和，计算左侧前缀和的值与整个数组的值减去当前前缀和的值
// 当两者相等时，返回当前索引

func pivotIndex(nums []int) int{
    sum, total := 0, 0
    for _, v := range nums {
        total += v
    }

    for i, val := range nums {
        sum += val
        if sum - nums[i] == total - sum {
            return i
        }
    }
    return -1
}
